app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetWeightLifting.kt

```diff
@@ -0,0 +1,222 @@
+                        weight = weightValue,
+                        weightType = weightType,
+                        index = editWorkoutSetWeightLiftingModel.index
+                    )
+                )
+            )
+        }
+    }
+
+    EditWorkoutSetsSection(
+        isErrorVisible = isSetErrorVisible,
+        editSetsValue = { editWorkoutFunction: EditWorkoutFunction, sets: String ->
+            isSetErrorVisible = false
+            try {
+                if (sets.isDigitsOnly() && sets != "0" && sets.isNotEmpty()) {
+                    var setsInt = sets.toIntOrNull()
+                    if (setsInt == null) {
+                        isSetErrorVisible = true
+                        return@EditWorkoutSetsSection
+                    }
+                    when (editWorkoutFunction) {
+                        EditWorkoutFunction.ADD_VALUE -> setsInt += 1
+                        EditWorkoutFunction.SUBTRACT_VALUE -> setsInt -= 1
+                    }
+                    setValue = setsInt.toString()
+                } else if (sets == "0" || sets.isEmpty()) {
+                    setValue = when (editWorkoutFunction) {
+                        EditWorkoutFunction.ADD_VALUE -> "1"
+                        EditWorkoutFunction.SUBTRACT_VALUE -> "0"
+                    }
+                }
+                if (setValue == "0") {
+                    isSetErrorVisible = true
+                }
+            } catch (e: Exception) {
+                isSetErrorVisible = true
+            }
+        },
+        setsValue = setValue,
+        onSetValueChange = {
+            isSetErrorVisible = it.isBlank()
+            setValue = it
+        }
+    )
+    EditWorkoutRepsSection(
+        isRepsErrorVisible = isRepErrorVisible,
+        editRepValue = { editWorkoutFunction: EditWorkoutFunction, reps: String ->
+            isRepErrorVisible = false
+            try {
+                if (reps.isDigitsOnly() && reps != "0" && reps.isNotEmpty()) {
+                    var repsInt = reps.toIntOrNull()
+                    if (repsInt == null) {
+                        isRepErrorVisible = true
+                        return@EditWorkoutRepsSection
+                    }
+                    when (editWorkoutFunction) {
+                        EditWorkoutFunction.ADD_VALUE -> repsInt += 1
+                        EditWorkoutFunction.SUBTRACT_VALUE -> repsInt -= 1
+                    }
+                    repValue = repsInt.toString()
+                } else if (reps == "0" || reps.isEmpty()) {
+                    repValue = when (editWorkoutFunction) {
+                        EditWorkoutFunction.ADD_VALUE -> "1"
+                        EditWorkoutFunction.SUBTRACT_VALUE -> "0"
+                    }
+                }
+                if (repValue == "0") {
+                    isRepErrorVisible = true
+                }
+            } catch (e: Exception) {
+                isRepErrorVisible = true
+            }
+        },
```

Comment:
This is duplicate code. Can we extract this out into a helper function and reuse it?